src/include/meson.build

```diff
@@ -0,0 +1,9 @@
+configure_file(output: 'config.h', configuration: cdata)
```

Comment:
Just a question, why is this in it's own directory?

Reply:
It's been a few days, I may misremember something, but I believe it goes like this.

With default settings, meson adds current source and build directories to include path.

They go right to the front of compiler's arguments: `cc -I. -I../src/foobar …`.

Under MinGW, `unistd.h` includes `process.h`. What it wants is to include the system header (`/mingw64/include/process.h`), but because of the argument order it actually includes `src/process.h` from tinc's sources, so we get an include cycle, and a compiler error.

Here's an [example test run](https://github.com/hg/tinc/runs/5611493644?check_suite_focus=true#step:7:142).

I [tried to fix it](https://github.com/hg/tinc/commit/068b1399cc78e25335ad5f4e59a622f772d02bc5) by renaming `src/process.h`, but thought that to be a pretty poor solution — who knows what conflicts we may get in the future, or on other operating systems?

So we have to disable this behavior for all targets with `implicit_include_directories: false`. From looking at other projects, I believe it's being used quite widely.

Generated headers go into the build directory. To be able to access them, you either have to use `implicit_include_directories: true` (which is the default), or `include_directories('.')`, which (if you put it into `src/meson.build`) pretty much overrides that setting and you get the `process.h` issue right back.

This is why it's moved into a separate subdirectory named `include`, which only contains the two generated headers and can be added to include path without adding other stuff (like `src/process.h`) that conflicts with system headers.

Thanks for the review. I'll fix everything else soon.